### PR TITLE
Convert WPComRestResponse to Kotlin

### PR DESCRIPTION
**Pull request by malinajirka**
Converts PostWPComRestResponse to Kotlin
- adds `@SerializedName` to all fields so we can for example safely rename our fields
- renames the fields so their names don't contain `_`

All fields but `PostsResponse.posts` and `PostWPComRestResponse.type` are nullable. These two fields would have been failing on NPE even before so I believe it's safe to make them explicitly NotNull.

To test:
1. Run `ReleaseStack_PostTestWPCom` and `ReleaseStack_PostTestXMLRPC`

### CODE REVIEW

File: fluxc/src/main/java/org/wordpress/android/fluxc/network/rest/wpcom/post/PostRestClient.java [hunk 1]
```diff
@@ -326,54 +327,54 @@ public void onErrorResponse(@NonNull WPComGsonNetworkError error) {
...
-        post.setSlug(from.slug);
-        post.setStatus(from.status);
-        post.setPassword(from.password);
-        post.setIsPage(from.type.equals("page"));
-
-        if (from.post_thumbnail != null) {
-            post.setFeaturedImageId(from.post_thumbnail.ID);
+        post.setRemotePostId(from.getRemotePostId());
+        post.setRemoteSiteId(from.getRemoteSiteId());
+        post.setLink(from.getUrl()); // Is this right?
```
[the remark this answers is not in the document]
Comment (malinajirka) [reply fluxc/src/main/java/org/wordpress/android/fluxc/network/rest/wpcom/post/PostRestClient.java]: Tbh I'd rather not touch the PostModel in this PR as it'd be a breaking change for all the repos using FluxC.

File: fluxc/src/main/java/org/wordpress/android/fluxc/network/rest/wpcom/post/PostRestClient.java [hunk 2]
```diff
@@ -326,54 +327,54 @@ public void onErrorResponse(@NonNull WPComGsonNetworkError error) {
...
+            post.setHasCapabilityPublishPost(from.getCapabilities().getPublishPost());
+            post.setHasCapabilityEditPost(from.getCapabilities().getEditPost());
+            post.setHasCapabilityDeletePost(from.getCapabilities().getDeletePost());
         }
 
-        if (from.parent != null) {
-            post.setParentId(from.parent.ID);
-            post.setParentTitle(from.parent.title);
+        if (from.getParent() != null) {
+            post.setParentId(from.getParent().ID);
```
[the remark this answers is not in the document]
Comment (malinajirka) [reply fluxc/src/main/java/org/wordpress/android/fluxc/network/rest/wpcom/post/PostRestClient.java]: Good point - fixed in f08f94d9a1d45740c9db982d67df42dc784fc10f

File: fluxc/src/main/java/org/wordpress/android/fluxc/network/rest/wpcom/post/PostRestClient.java [hunk 3]
```diff
@@ -326,54 +327,54 @@ public void onErrorResponse(@NonNull WPComGsonNetworkError error) {
...
+        if (from.getPostThumbnail() != null) {
+            post.setFeaturedImageId(from.getPostThumbnail().getId());
         }
-        post.setPostFormat(from.format);
-        if (from.geo != null) {
-            post.setLatitude(from.geo.latitude);
-            post.setLongitude(from.geo.longitude);
+        post.setPostFormat(from.getFormat());
+        if (from.getGeo() != null) {
+            post.setLatitude(from.getGeo().latitude);
```
[the remark this answers is not in the document]
Comment (malinajirka) [reply fluxc/src/main/java/org/wordpress/android/fluxc/network/rest/wpcom/post/PostRestClient.java]: Good point - fixed in f08f94d9a1d45740c9db982d67df42dc784fc10f

**Comment by malinajirka** [comment]
Thanks @maxme! I've converted GeoLocation and PostParent to Kotlin. I have decided not to update the PostModel as part if this PR, if you don't mind. 

It's ready for another round.

**Comment by maxme** [comment]
LGTM, :shipit:

**Pull request merged**